### PR TITLE
docs: add kubernetes compatibilit

### PR DESCRIPTION
**Pull request by huiwq1990**
Signed-off-by: huiwq1990 <huiwq1990@163.com>

Fixes https://github.com/argoproj/argo-workflows/issues/8198

Please do not open a pull request until you have checked ALL of these:

* [x] Create the PR as draft .
* [ ] Run `make pre-commit -B` to fix codegen and lint problems.
* [x] Sign-off your commits (otherwise the DCO check will fail).
* [x] Use [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) (otherwise the commit message check will fail).
* [x] "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
* [ ] Add unit or e2e tests. Say how you tested your changes. If you changed the UI, attach screenshots.
* [ ] Github checks are green.
* [ ] Once required tests have passed, mark your PR "Ready for review".

If changes were requested, and you've made them, dismiss the review to get it reviewed again.

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -35,3 +35,22 @@ Otherwise, we typically release every two weeks:
 
 * Patch fixes for the current stable version.
 * The next release candidate, if we are currently in a release-cycle.
+
+## Kubernetes Compatibility Matrix
+
+| Argo Workflows \ Kubernetes | 1.17 | 1.18 | 1.19 | 1.20 | 1.21 | 1.22 | 1.23 |
```
Comment (tooptoop4) [inline docs/releases.md:41]: 1.24?

Comment (yolu-kxs) [reply docs/releases.md:41]: 1.25?

Comment (matt-carr) [reply docs/releases.md:41]: https://github.com/argoproj/argo-workflows/commit/36646ef81cb4775c1ef31861f01331bc75166e7b

Comment (huiwq1990) [reply docs/releases.md:41]: @tooptoop4 @yolu-kxs did we support 1.24 and 1.25, I did not test on them?

Comment (yolu-kxs) [reply docs/releases.md:41]: I am not sure if it supports 1.24 and 1.25. Has someone tested it?

Comment (Sushant20) [reply docs/releases.md:41]: I have argo workflows 3.4.4 installed on k8s 1.24, it works as expected

**Review by sarabala1979 — APPROVED** [review]

**Comment by sarabala1979** [comment]
@huiwq1990 Can you update new v3.4.4 will support v1.24

**Pull request merged**